### PR TITLE
quick combat vs ai

### PR DESCRIPTION
**Pull request by Laserlicht**
continue of #2744

fixes https://github.com/vcmi/vcmi/issues/2671

-> No green messages anymore

Can some people test it?

### CODE REVIEW

**Comment by krs0** [comment]
I have tested it in some scenarios:

Looks ok in single player. I can quick battle neutrals and neutral heroes
Works on heroes with Tactics as well.
Looks ok on hot seat... did not kick in when battling heroes.
AutoCombat works ok in all the above scenarios
No system or console messages 

I did not find an option for: Imediate Finish with Autocombat (Q shortcut in HD mod), do not recall if this was implemented in VCMI or not.
Multiple replays still not an option :p

**Comment by IvanSavenko** [comment]
Can you confirm that if you are attacked by AI and lose, AI will not resume actions, e.g. movement until you close battle results windows? For example, keep an eye on console - there should be no messages from AI and AI should not end turn while battle results are open on your side.

Similarly - get attacked and lose, don't close battle results for a bit, then click replay.

I think these cases were main source of problems before - AI resumed acting before player decided whether to replay battle.

**Comment by krs0** [comment]
Provoked the AI to attack me in 3-4 battles and lost them. Used a mix of Quick battle and AutoCombat.

No AI messages / actions while windows were still on screen.
Once results applied, AI continued with its turn as expected.

**Comment by Laserlicht** [comment]
Thanks for testing. :)

Every green message I've got in the original pull request comes from the here removed lines of the AI's.

In my final testings I've never got green messages again.

@krs0 You can do multiple replays. Just start with them in "Extra option" in scenario options.
For finishing with auto combat see this PR: #3569

**Review by IvanSavenko — COMMENTED** [review]
And just in case - can somebody confirm that AI-vs-AI battles also work fine?

File: client/Client.cpp [hunk 1]
```diff
@@ -585,14 +585,22 @@ void CClient::battleStarted(const BattleInfo * info)
...
+		auto endTacticPhaseIfEligible = [info](const CPlayerInterface * interface)
 		{
-			auto side = att->cb->getBattle(info->battleID)->playerToSide(att->playerID);
-			auto action = BattleAction::makeEndOFTacticPhase(*side);
-			att->cb->battleMakeTacticAction(info->battleID, action);
-		}
+			if (interface->cb->getBattle(info->battleID)->battleGetTacticDist())
+			{
+				auto side = interface->cb->getBattle(info->battleID)->playerToSide(interface->playerID);
+				auto action = BattleAction::makeEndOFTacticPhase(*side);
```
[the remark this answers is not in the document]
Comment (Laserlicht) [reply client/Client.cpp]: You mean like this? (see commit)

File: AI/Nullkiller/AIGateway.cpp [hunk 2]
```diff
@@ -1131,15 +1131,6 @@ void AIGateway::battleEnd(const BattleID & battleID, const BattleResult * br, Qu
...
 
-	if (queryID != QueryID::NONE)
-	{
-		status.addQuery(queryID, "Combat result dialog");
-		const int confirmAction = 0;
-		requestActionASAP([=]()
-		{
-			answerQuery(queryID, confirmAction);
-		});
-	}
```
[the remark this answers is not in the document]
Comment (Laserlicht) [reply AI/Nullkiller/AIGateway.cpp:1142]: This code was added by #1108

Seems for me like dead code. But if we do auto combat between two heroes this code triggers the green messages from @dydzio0614 PR #2744

Without it everything I tested worked fine without triggering green messages.

**Comment by Laserlicht** [comment]
> And just in case - can somebody confirm that AI-vs-AI battles also work fine?

Tested it on a custom maps with many heroes and players killing themself. Seems okay.

**Pull request merged**